src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs

```diff
@@ -53,6 +55,54 @@ public void BufferSize()
             Assert.Equal(1234, bufferedStream.BufferSize);
         }
 
+        [ConditionalTheory(nameof(IsX64))]
+        [OuterLoop]
+        [InlineData(int.MaxValue / 2 + 1)]
+        public void WriteFromByte_InputSizeLargerThanHalfOfMaxInt_ShouldSuccess(int inputSize)
+        {
+            byte[] bytes;
+
+            try
+            {
+				bytes = new byte[inputSize];
+            }
+            catch (OutOfMemoryException)
+            {
+                return;
+            }
+
+			var writableStream = new WriteOnlyStream();
+            using (var bs = new BufferedStream(writableStream))
+            {
+                bs.Write(bytes, 0, inputSize);
+                Assert.Equal(inputSize, writableStream.GetPosition());
+            }
+        }
```

Comment:
Nit: the indentation in this method looks off

Reply:
Fixed.